# Walidacja: warianty slice-aware (bestof5 / qfserve / sliceaware) (Sprint 4+)

## Cel
Warianty slice-aware byly oceniane tylko na pojedynczym tescie (gdzie bestof5 dawal +2.37, qfserve
+2.20). Tu sprawdzamy je UCZCIWIE -- walk-forward przez kilka sezonow + McNemar. Baseline liczony
raz na rok (cache), warianty go reuzywaja.

In [1]:
import sys
from pathlib import Path
sys.path.insert(0, str(Path("../src").resolve()))

## Uruchomienie (od zera) -- baseline + 3 warianty na kazdy sezon

In [2]:
import tennis_model_validate_variants as m
m.main()


===== ROK 2020 =====


  baseline match=0.6176 (n=272)


    sliceaware   match=0.5919  delta=-0.0257


    bestof5_v1   match=0.6213  delta=+0.0037


    qfserve_v3   match=0.6066  delta=-0.0110



===== ROK 2021 =====


  baseline match=0.6667 (n=522)


    sliceaware   match=0.6762  delta=+0.0096


    bestof5_v1   match=0.6705  delta=+0.0038


    qfserve_v3   match=0.6533  delta=-0.0134



===== ROK 2022 =====


  baseline match=0.6728 (n=547)


    sliceaware   match=0.6782  delta=+0.0055


    bestof5_v1   match=0.6709  delta=-0.0018


    qfserve_v3   match=0.6746  delta=+0.0018



===== ROK 2023 =====


  baseline match=0.6346 (n=561)


    sliceaware   match=0.6239  delta=-0.0107


    bestof5_v1   match=0.6364  delta=+0.0018


    qfserve_v3   match=0.6168  delta=-0.0178



===== ROK 2024 =====


  baseline match=0.6186 (n=590)


    sliceaware   match=0.6169  delta=-0.0017


    bestof5_v1   match=0.6322  delta=+0.0136


    qfserve_v3   match=0.6237  delta=+0.0051



===== ROK 2025 =====


  baseline match=0.6566 (n=530)


    sliceaware   match=0.6528  delta=-0.0038


    bestof5_v1   match=0.6679  delta=+0.0113


    qfserve_v3   match=0.6358  delta=-0.0208



WALK-FORWARD: warianty slice-aware vs baseline

--- sliceaware ---
 year  baseline  variant   delta
 2020    0.6176   0.5919 -0.0257
 2021    0.6667   0.6762  0.0096
 2022    0.6728   0.6782  0.0055
 2023    0.6346   0.6239 -0.0107
 2024    0.6186   0.6169 -0.0017
 2025    0.6566   0.6528 -0.0038
  POOLED (3022 meczow): baseline=0.6463  sliceaware=0.6436  delta=-0.0026  (dodatnie 2/6 lat)
  McNemar: b=98 c=90 z=0.51 p=0.6097 -> brak istotnosci

--- bestof5_v1 ---
 year  baseline  variant   delta
 2020    0.6176   0.6213  0.0037
 2021    0.6667   0.6705  0.0038
 2022    0.6728   0.6709 -0.0018
 2023    0.6346   0.6364  0.0018
 2024    0.6186   0.6322  0.0136
 2025    0.6566   0.6679  0.0113
  POOLED (3022 meczow): baseline=0.6463  bestof5_v1=0.6519  delta=+0.0056  (dodatnie 5/6 lat)
  McNemar: b=114 c=131 z=1.02 p=0.3067 -> brak istotnosci

--- qfserve_v3 ---
 year  baseline  variant   delta
 2020    0.6176   0.6066 -0.0110
 2021    0.6667   0.6533 -0.0134
 2022    0.6728   0.6746  0.0

## Wnioski
Na walk-forward (2020-2025, N=3022) ZADEN wariant slice-aware nie rozni sie istotnie od baseline --
wszystkie McNemar p > 0.05 (najnizsze qfserve p=0.076). Dawne "spektakularne" wyniki z pojedynczego
testu (bestof5 +2.37, qfserve +2.20 p.p.) rozplywaja sie w szumie: pooled bestof5 +0.56 p.p.
(nieistotne), qfserve -0.89 p.p. (kierunkowo gorszy), sliceaware -0.26 p.p. **Zaden wariant nie
pobija glownego modelu `tennis_model.py` w sposob istotny.** Pojedynczy test set klamie -- przewaga
z jednego sezonu nie utrzymuje sie na wielu.